scipy/optimize/_linprog_util.py

```diff
@@ -729,19 +741,34 @@ def where(A):
         b_eq = b_eq - A_eq[:, i_f].dot(lb[i_f])
         b_ub = b_ub - A_ub[:, i_f].dot(lb[i_f])
         c = c[i_nf]
+        x_undo = lb[i_f]  # not x[i_f], x is just zeroes
         x = x[i_nf]
         # user guess x0 stays separate from presolve solution x
         if x0 is not None:
             x0 = x0[i_nf]
         A_eq = A_eq[:, i_nf]
         A_ub = A_ub[:, i_nf]
-        # record of variables to be added back in
-        undo = [np.nonzero(i_f)[0], lb[i_f]]
-        # don't remove these entries from bounds; they'll be used later.
-        # but we _also_ need a version of the bounds with these removed
+        # modify bounds
         lb_mod = lb[i_nf]
         ub_mod = ub[i_nf]
 
+        def rev(x_mod):
+            # Function to restore x: insert x_undo into x_mod.
+            # When elements have been removed at positions k1, k2, k3, ...
+            # then these must be replaced at (after) positions k1-1, k2-2,
+            # k3-3, ... in the modified array to recreate the original
+            i = np.flatnonzero(i_f)
+            # Number of variables to restore
+            N = len(i)
+            index_offset = list(range(N))
```

Comment:
Why not `np.arange(N)` so you can just do `i - index_offset` below?

Reply:
Yes, why not, it is much cleaner.